overlord/booted.go

```diff
@@ -57,22 +60,56 @@ func SyncBoot() error {
 	kernelSnap, _ := bootloader.GetBootVar("snappy_kernel")
 	osSnap, _ := bootloader.GetBootVar("snappy_os")
```

Comment:
Can we rename these variables while we're here?  `kernel_snap` and `core_snap`? In fact, how about generalizing the logic so we have a single boot env variable `snaps`, comma separated? That's less churning with foreign systems, less non-atomic changes, and the logic below seems appropriate for that.